src/main/java/spring/backend/core/util/EmailUtil.java

```diff
@@ -0,0 +1,30 @@
+package spring.backend.core.util;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+import spring.backend.core.dto.request.SendEmailRequest;
+
+@Component
+public class EmailUtil {
+    @Value("${spring.mail.sender}")
+    private String sender;
+
+    private final JavaMailSender mailSender;
+
+    @Autowired
+    public EmailUtil(JavaMailSender mailSender) {
+        this.mailSender = mailSender;
+    }
+
+    public void send(SendEmailRequest sendEmailRequest) {
+        SimpleMailMessage message = new SimpleMailMessage();
+        message.setFrom(sender);
+        message.setTo(sendEmailRequest.to());
+        message.setSubject(sendEmailRequest.subject());
+        message.setText(sendEmailRequest.text());
+        mailSender.send(message);
+    }
```

Comment:
```
	/**
	 * Send the given simple mail message.
	 * @param simpleMessage the message to send
	 * @throws MailParseException in case of failure when parsing the message
	 * @throws MailAuthenticationException in case of authentication failure
	 * @throws MailSendException in case of failure when sending the message
	 */
	default void send(SimpleMailMessage simpleMessage) throws MailException {
		send(new SimpleMailMessage[] {simpleMessage});
	}
```

java 공식 문서인데 MailParseException, MailAuthenticationException, MailSendException 세 개의 에러가 발생할 수 있는데 위 에러는 전부 MailException을 상속받고 있기 때문에 MailException에 대해 try, catch가 필요해 보입니다.

세 종류의 에러가 발생할 수 있지만, 결론은 메일 전송에 대한 에러이기 때문에 MailException 만 처리하고 로깅을 남겨도 좋을 거 같습니다!

```java
// 로깅 안할 시
public void send(SendEmailRequest sendEmailRequest) throws MailException {
        SimpleMailMessage message = new SimpleMailMessage();
        message.setFrom(sender);
        message.setTo(sendEmailRequest.to());
        message.setSubject(sendEmailRequest.subject());
        message.setText(sendEmailRequest.text());
        mailSender.send(message);
    }
 ```

``` java
// 로깅할 시
 public void send(SendEmailRequest sendEmailRequest) {
        try {
            SimpleMailMessage message = new SimpleMailMessage();
            message.setFrom(sender);
            message.setTo(sendEmailRequest.to());
            message.setSubject(sendEmailRequest.subject());
            message.setText(sendEmailRequest.text());
            mailSender.send(message);
        } catch (MailException e) {
            log.warn("[EmailUtil] Error sending email to {} with subject: {}", sendEmailRequest.to(), sendEmailRequest.subject(), e);
            throw e;
        }
    }
```

이런 식으로요!

Reply:
에러를 세분화해서 관리해보겠습니다..!